### PR TITLE
Simplify test configuration

### PR DESCRIPTION
**Pull request by dkijania**
Refactored configuration and starter for jormungandr process. I think we went too far in hiding internals of bootstrap procedure for jormungandr. We introduced uber config which mixes secret, node and block0 configuration. It is hard to tell which setting is corresponding to which configuration part. I split this concept and move configuration in integration test. As a effect jormungandr-automation can be used in other projects like hersir or jorup as this is very open to defined many different configurations. However we still need to be able to define intent in integration test (rather than implementation details) and to satisfy this need i introduced SingleTestBootstrapper and others.

### CODE REVIEW

File: src/jormungandr/testing/jormungandr-automation/resources/explorer/graphql/allblocks.graphql [hunk 1]
```diff
@@ -3,160 +3,22 @@ query AllBlocks($last: Int!) {
...
-                                        denominator
-                                    }}
-    ... on PoolRewardParticipationCapping { min max }
-    ... on AddCommitteeId { addCommitteeId}
-    ... on RemoveCommitteeId { removeCommitteeId }
-    ... on PerVoteCertificateFee {
-                                    certificateVotePlan
-                                    certificateVoteCast
-                                }
-    ... on TransactionMaxExpiryEpochs { transactionMaxExpiryEpochs }
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-automation/resources/explorer/graphql/allblocks.graphql:161]: I think you are merging the old version of this and deleting the new, longer one

File: src/jormungandr/testing/jormungandr-automation/resources/explorer/graphql/blocksbychainlength.graphql [hunk 2]
```diff
@@ -1,155 +1,5 @@
...
-                                        numerator
-                                        denominator
-                                    }}
-    ... on PoolRewardParticipationCapping { min max }
-    ... on AddCommitteeId { addCommitteeId}
-    ... on RemoveCommitteeId { removeCommitteeId }
-    ... on PerVoteCertificateFee {
-                                    certificateVotePlan
-                                    certificateVoteCast
-                                }
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-automation/resources/explorer/graphql/blocksbychainlength.graphql:153]: same as above

File: src/jormungandr/testing/jormungandr-automation/src/jormungandr/explorer/client.rs [hunk 3]
```diff
@@ -43,7 +43,7 @@ impl GraphQlClient {
         if self.print_out {
             println!(
                 "running query: {:#?}, against: {}",
-                query.operation_name, self.base_url
+                query.query, self.base_url
             );
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-automation/src/jormungandr/explorer/client.rs:47]: same as above

File: src/jormungandr/testing/jormungandr-automation/src/jormungandr/explorer/mod.rs [hunk 4]
```diff
@@ -149,29 +149,6 @@ impl ExplorerProcess {
...
-        }
-
-        if wait.timeout_reached() {
-            println!("Timeout reached");
-            false
-        } else {
-            println!("Explorer is up");
-            true
-        }
-    }
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-automation/src/jormungandr/explorer/mod.rs:174]: same as above

File: src/jormungandr/testing/jormungandr-automation/src/jormungandr/explorer/verifiers/mod.rs [hunk 5]
```diff
@@ -125,16 +121,9 @@ impl ExplorerVerifier {
...
     fn decode_bech32_pk(bech32_public_key: &str) -> PublicKey<Ed25519> {
         let (_, data, _variant) = bech32::decode(bech32_public_key).unwrap();
         let dat = Vec::from_base32(&data).unwrap();
         PublicKey::<Ed25519>::from_binary(&dat).unwrap()
     }
-
-    pub fn decode_block(encoded_block: String) -> Block {
-        let bytes_block = hex::decode(encoded_block.trim()).unwrap();
-        let reader = std::io::Cursor::new(&bytes_block);
-        Block::deserialize(&mut Codec::new(reader)).unwrap()
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-automation/src/jormungandr/explorer/verifiers/mod.rs:138]: same as above

**Comment by dkijania** [comment]
thanks @kukkok3. I've reverted those changes

File: src/jormungandr/explorer/src/db/indexing.rs [hunk 6]
```diff
@@ -170,28 +170,6 @@ impl ExplorerBlock {
...
-                        Some(ExplorerTransaction::from(
-                            &context,
-                            &fragment_id,
-                            &tx,
-                            Some(Certificate::UpdateVote(tx.payload().into_payload())),
-                            offset,
-                            &current_block_txs,
-                        ))
-                    }
                     Fragment::Transaction(tx) => {
```
Comment (kukkok3) [inline src/jormungandr/explorer/src/db/indexing.rs:173]: This also needs to be reverted

File: src/jormungandr/testing/mjolnir/src/mjolnir_lib/generators/fragment_generator.rs [hunk 7]
```diff
@@ -259,11 +259,13 @@ impl<'a, S: SyncNode + Send> FragmentGenerator<'a, S> {
...
                 let index = self.rand.next_u32() as usize % self.active_stake_pools.len();
                 let stake_pool = self.active_stake_pools.get(index).unwrap();
+
                 self.fragment_sender
                     .send_owner_delegation(&mut self.sender, stake_pool, &self.node)
             }
             4 => {
                 let stake_pool = StakePool::new(&self.sender);
+                self.active_stake_pools.push(stake_pool.clone());
                 self.fragment_sender.send_pool_registration(
```
Comment (kukkok3) [inline src/jormungandr/testing/mjolnir/src/mjolnir_lib/generators/fragment_generator.rs:269]: this needs to be reverted

File: src/jormungandr/testing/jormungandr-integration-tests/src/jormungandr/explorer/block.rs [hunk 8]
```diff
@@ -184,345 +130,88 @@ pub fn explorer_block_test() {
...
+        assert!(response
+            .unwrap()
+            .errors
+            .unwrap()
+            .first()
+            .unwrap()
+            .message
+            .contains(error_message));
+    }
 }
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-integration-tests/src/jormungandr/explorer/block.rs:217]: I think in this file all_block_test and block_by_chain_lenght test are gone missing

**Comment by dkijania** [comment]
rebased and updated the commits

**Pull request merged**